ToDoList-Redux/Views/CheckmarkCellView.swift

```diff
@@ -19,22 +19,32 @@ struct CheckmarkCellView: View {
             Image(systemName: viewModel.state.imageName)
                 .resizable()
                 .frame(width: 20, height: 20)
-                .onTapGesture { viewModel.dispatch(.toggle(viewModel.state.id))}
+                .onTapGesture { viewModel.dispatch(.toggle) }
             Text(viewModel.state.title)
         }
     }
 }
 
+#if DEBUG
 struct CheckmarkCellView_Previews: PreviewProvider {
+    static let stateMock = AppState.mock
+    static let mockStore = ObservableViewModel<AppAction, AppState>.mock(
+        state: stateMock,
+        action: { action, _, state in
+            state = Reducer.app.reduce(action, state)
```

Comment:
We can use all reducers here! The preview will behave very close to the real app, but without the Middleware events or side-effects. If you want to emulate the middleware side-effects callbacks you still can do it in this closure here.

Reply:
Note that this is one of my least favorite change 🤷‍♂️. While it seems great at first sight to be using the AppState mock, it's pushing down to our "independent" View some of the App level constructs.

Reply:
That's fair. We could pick only the reducers that are relevant and that way use a "store" from that level.